ow_lander/scripts/all_action_trajectories.py

```diff
@@ -0,0 +1,853 @@
+  goal_pose.orientation.w = goal_pose.orientation.w
+
+  # Ask the planner to generate a plan to the approximate joint values generated
+  # by kinematics builtin IK solver. For more insight on this issue refer to:
+  # https://github.com/nasa/ow_simulator/pull/60
+  if approximate:
+    move_arm.set_joint_value_target(goal_pose, True)
+  else:
+    move_arm.set_pose_target(goal_pose)
+
+  plan = move_arm.plan()
+  
+  if len(plan.joint_trajectory.points) == 0:  # If no plan found, abort
+    return False
+
+  return plan  
+
+
+def go_to_Z_coordinate(move_arm, cs, goal_pose, x_start, y_start, z_start, approximate=True):
+  """
+  :param approximate: use an approximate solution. default True
+  :type move_group: class 'moveit_commander.move_group.MoveGroupCommander'
+  :type x_start: float
+  :type y_start: float
+  :type z_start: float
+  :type approximate: bool
+  """
+  
+  move_arm.set_start_state(cs)
+  goal_pose.position.z = z_start
+  
+  goal_pose.orientation.x = goal_pose.orientation.x
+  goal_pose.orientation.y = goal_pose.orientation.y
+  goal_pose.orientation.z = goal_pose.orientation.z
+  goal_pose.orientation.w = goal_pose.orientation.w
+
+  # Ask the planner to generate a plan to the approximate joint values generated
+  # by kinematics builtin IK solver. For more insight on this issue refer to:
+  # https://github.com/nasa/ow_simulator/pull/60
+  if approximate:
+    move_arm.set_joint_value_target(goal_pose, True)
+  else:
+    move_arm.set_pose_target(goal_pose)
+
+  plan = move_arm.plan()
+  
+  
+  if len(plan.joint_trajectory.points) == 0:  # If no plan found, abort
+    return False
+
+  return plan
+
+def move_to_pre_trench_configuration_dig_circ(move_arm, robot, x_start, y_start):
+  """
+  :type move_arm: class 'moveit_commander.move_group.MoveGroupCommander'
+  :type x_start: float
+  :type y_start: float
+  """
+  # Initilize to current position
+  joint_goal = move_arm.get_current_pose().pose
+  robot_state = robot.get_current_state()
+  move_arm.set_start_state(robot_state)
+  
+  # Compute shoulder yaw angle to trench
+  alpha = math.atan2(y_start-constants.Y_SHOU, x_start-constants.X_SHOU)
+  h = math.sqrt(pow(y_start-constants.Y_SHOU, 2) +
+                pow(x_start-constants.X_SHOU, 2))
+  l = constants.Y_SHOU - constants.HAND_Y_OFFSET
+  beta = math.asin(l/h)
+  # Move to pre trench position, align shoulder yaw
+  joint_goal = move_arm.get_current_joint_values()
+  joint_goal[constants.J_DIST_PITCH] = 0.0
+  joint_goal[constants.J_HAND_YAW] = 0.0
+  joint_goal[constants.J_PROX_PITCH] = -math.pi/2
+  joint_goal[constants.J_SHOU_PITCH] = math.pi/2
+  joint_goal[constants.J_SHOU_YAW] = alpha + beta
+
+  # If out of joint range, abort
+  if (is_shou_yaw_goal_in_range(joint_goal) == False):
+    return False
+
+  joint_goal[constants.J_SCOOP_YAW] = 0
+  
+  plan = move_arm.plan(joint_goal)
+  return plan
+
+
+def dig_circular(move_arm, move_limbs, robot, moveit_fk, args):
```

Comment:
This function is quite long.  Perhaps break it into two, for parallel and non-parallel?

Reply:
parallel is an argument .. it is same implementation as services.... so making this change.